server/src/main/java/org/opensearch/index/engine/ReadOnlyEngine.java

```diff
@@ -93,9 +93,24 @@ public class ReadOnlyEngine extends Engine {
     private final CompletionStatsCache completionStatsCache;
     private final boolean requireCompleteHistory;
     private final TranslogManager translogManager;
+    private final Version minimumSupportedVersion;
 
     protected volatile TranslogStats translogStats;
 
+    /**
+     * Wrapper constructor that turns off support for reading old indices.
+     */
+    public ReadOnlyEngine(
+        EngineConfig config,
+        SeqNoStats seqNoStats,
+        TranslogStats translogStats,
+        boolean obtainLock,
+        Function<DirectoryReader, DirectoryReader> readerWrapperFunction,
+        boolean requireCompleteHistory
+    ) {
+        this(config, seqNoStats, translogStats, obtainLock, readerWrapperFunction, requireCompleteHistory, null);
```

Comment:
```suggestion
        this(config, seqNoStats, translogStats, obtainLock, readerWrapperFunction, requireCompleteHistory, Version.MIN_SUPPORTED_MAJOR);
```

Reply:
I think usage of `null` here is very confusing: the `minimumSupportedVersion` is not going anywhere, it is `Version.MIN_SUPPORTED_MAJOR`